### ISSUE TITLE
TypeError: Slice Object is not Iterable

### ISSUE THREAD

**Issue by GoogleCodeExporter** [abulka/pynsource#34]
```
What steps will reproduce the problem?
1. download and install pynsource 1.61 (or standalone)
2. download matplotlib source code from github
3. import matplotlib/lib/matplotlib/color.py (or any other code in the folder) 
to pynsourceGUI

What is the expected output? What do you see instead?
Expected output is a nice UML diagram.
Instead, getting an error msg "TypeError: Slice Object is not iterable"

What version of the product are you using? On what operating system?
Tried both 1.61 and 1.60 on windows7

Please provide any additional information below.
There is no 'Slice' object in the code that I have tried.
So really don't know where it is coming from.
I just tried more code from the same folder, some files work and some gives 
error msg like that.
This does not happen in older version

```

Original issue reported on code.google.com by `socialap...@gmail.com` on 25 Jan 2013 at 9:20

**Comment by abulka** [comment]
There is actually a lot of extended slicing going on in that file e.g. 

    grep "\[" colors.py |grep "\:"

See info on extended slicing [here](https://stackoverflow.com/questions/509211/understanding-pythons-slice-notation).

This error can be fixed by changing the parser code in `src/parsing/core_parser_ast.py` to add the word `enumerate`:

```
    def visit_ExtSlice(self, node):
        for idx, item in enumerate(node.dims):  # <--- added enumerate
            if idx:
                self.write(", ")
            self.visit(item)
```

this fix has not been incorporated into the codebase yet.

**Comment by abulka** [comment]
Should be now fixed.

**Comment by abulka** [comment]
fixed in commit 267ccae235c032374dc9b47edcfca9b290f277eb
